server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -250,6 +252,7 @@ public ThreadPool(
             Names.TRANSLOG_TRANSFER,
             new ScalingExecutorBuilder(Names.TRANSLOG_TRANSFER, 1, halfProcMaxAt10, TimeValue.timeValueMinutes(5))
         );
+        builders.put(Names.TRANSLOG_SYNC, new FixedExecutorBuilder(settings, Names.TRANSLOG_SYNC, allocatedProcessors * 4, 10000));
```

Comment:
What is the deciding factor on threadpool size? Indexing throughput or number of shards?

Reply:
This has been set to higher numbers to achieve maximum throughput for higher number of shards. This can be tuned basis the performance benchmarking. I am planning to run OSB and find out the choking point in the system.